path/basename_test.ts

```diff
@@ -4,33 +4,94 @@
 import { assertEquals } from "../testing/asserts.ts";
 import * as path from "./mod.ts";
 
+// Test suite from "GNU core utilities"
+// https://github.com/coreutils/coreutils/blob/master/tests/misc/basename.pl
+const COREUTILS_TESTSUITE = [
+  [["d/f"], "f"],
+  [["/d/f"], "f"],
+  [["d/f/"], "f"],
+  [["d/f//"], "f"],
+  [["f"], "f"],
+  [["/"], "/"],
+  [["///"], "/"],
+  [["///a///"], "a"],
+  [[""], ""],
+  [["aa", "a"], "a"],
+  [["a-a", "-a"], "a"],
+  [["f.s", ".s"], "f"],
+  [["fs", "s"], "f"],
+  [["fs", "fs"], "fs"],
+  [["fs/", "s"], "f"],
+  [["dir/file.suf", ".suf"], "file"],
+  [["fs", "x"], "fs"],
+  [["fs", ""], "fs"],
+  [["fs/", "s/"], "fs"],
+];
+
+const POSIX_TESTSUITE = [
+  [[""], ""],
+  [["/dir/basename.ext"], "basename.ext"],
+  [["/basename.ext"], "basename.ext"],
+  [["basename.ext"], "basename.ext"],
+  [["basename.ext/"], "basename.ext"],
+  [["basename.ext//"], "basename.ext"],
+  [["aaa/bbb", "/bbb"], "bbb"],
+  [["aaa/bbb", "a/bbb"], "bbb"],
+  [["aaa/bbb", "bbb"], "bbb"],
+  [["aaa/bbb//", "bbb"], "bbb"],
+  [["aaa/bbb", "bb"], "b"],
+  [["aaa/bbb", "b"], "bb"],
+  [["/aaa/bbb", "/bbb"], "bbb"],
+  [["/aaa/bbb", "a/bbb"], "bbb"],
+  [["/aaa/bbb", "bbb"], "bbb"],
+  [["/aaa/bbb//", "bbb"], "bbb"],
+  [["/aaa/bbb//", "a/bbb"], "bbb"],
+  [["/aaa/bbb", "bb"], "b"],
+  [["/aaa/bbb", "b"], "bb"],
+  [["/aaa/bbb"], "bbb"],
+  [["/aaa/"], "aaa"],
+  [["/aaa/b"], "b"],
+  [["/a/b"], "b"],
+  [["//a"], "a"],
+  [["///"], "/"],
+  [["///", "bbb"], "/"],
+  [["//", "bbb"], "/"],
+];
+
+const WIN32_TESTSUITE = [
+  [["\\dir\\basename.ext"], "basename.ext"],
+  [["\\basename.ext"], "basename.ext"],
+  [["basename.ext"], "basename.ext"],
+  [["basename.ext\\"], "basename.ext"],
+  [["basename.ext\\\\"], "basename.ext"],
+  [["foo"], "foo"],
+  [["aaa\\bbb", "\\bbb"], "bbb"],
+  [["aaa\\bbb", "a\\bbb"], "bbb"],
+  [["aaa\\bbb", "bbb"], "bbb"],
+  [["aaa\\bbb\\\\\\\\", "bbb"], "bbb"],
+  [["aaa\\bbb", "bb"], "b"],
+  [["aaa\\bbb", "b"], "bb"],
+  [["/aaa/bbb", "bb"], "b"],
+  [["C:"], "\\"],
```

Comment:
Should the result be `""`? The result of `posix.basename('')` is `""`.

Reply:
`C;` represents a relative path from the current directory of the C: drive. `C:\\` represents a root directory of the C: dirve. I grab it in https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats

Reply:
Ah, great catch, thanks! Updated